app/assets/javascripts/audits/batch-allocation.js

```diff
@@ -0,0 +1,38 @@
+window.GOVUKAdmin = window.GOVUKAdmin || {};
+window.GOVUKAdmin.Modules = window.GOVUKAdmin.Modules || {};
+
+(function (Modules) {
+  "use strict";
+
+  Modules.BatchAllocation = function () {
+    this.start = function ($root) {
+      this._initElements($root);
+      this._bindEventHandlers();
+    };
+
+    this._initElements = function ($root) {
+      this.$batchSizeInput = $root.find("#batch_size");
+      this.$checkboxes = $root.find("input[type=checkbox]");
+    };
+
+    this._bindEventHandlers = function () {
+      this.$batchSizeInput.keyup($.proxy(this.onBatchSizeChange, this));
+      this.$checkboxes.click($.proxy(this.onCheckboxClick, this));
+    };
+
+    this.onBatchSizeChange = function () {
+      this.$checkboxes
+        .prop('checked', false)
+        .slice(0, this.$batchSizeInput.val())
```

Comment:
(NB: users can input things like `-10` and this will select everything except the last 10 elements - probably not a common occurrence, but that's the behaviour we'd get here)